site/blog/2016/jul/1/sphinx-and-rtd-for-writers.rst

```diff
@@ -0,0 +1,548 @@
+* http://docs.readthedocs.io/en/latest/localization.html
+
+My blog
+-------
+
+Sphinx is quite versatile,
+which means you can use it for lots of different things.
+I use a package called ``ablog`` for hosting of my blog over at http://ericholscher.com.
+
+The simplest thing you do is specify that a document is a blog post with the ``post`` directive::
+
+    .. post:: 2016-03-15 09:00
+       :tags: writing, stc, sphinx
+
+This adds it to the set of posts that get shown in any document where you use::
+
+    .. postlist:: 
+       :tags: stc
+
+You can even create specific archive pages for categories or tags with ``postlist``,
+or just output all of your posts.
+
+This shows some of the magical things you can do with Sphinx's extensibility.
+If you're curious,
+this article was actually written in reStructuredText,
+and you can see it in full here: https://github.com/ericholscher/ericholscher.com/blob/master/site/blog/2016/jul/1/sphinx-and-rtd-for-writers.rst
+
+Custom builders
+---------------
+
+Along with the basic output formats,
+Sphinx also supports custom builders that do other tasks.
+This is a place where you can really be creative,
+and folks in the community have been.
+
+For example,
+these are a sampling of the builders that ship with Sphinx:
+
+* A linkcheck builder that tells you about broken URL's
+* A builder that outputs all the things changed in the latest version of your code for a changelog
+* An XML builder that outputs a representation of your documents in XML
+* A Man page builder that builds man pages from your documentation
+* JSON builder that outputs your pages as HTML inside of JSON, with some metadata, for embedding dynamically
+
+Trade offs with Sphinx
+----------------------
+
+Every tool has it's issues and limitations.
+I'd like to address some of the major issues with Sphinx,
+so that you can go into using it with eyes wide open.
+
+The biggest issue is that it is originally a *programmer tool*.
+This means that a lot of the designs assume knowledge of code,
+especially for installation and extending the tools.
+Knowledge of the command line is definitely required.
+
+The markup language,
+reStructuredText,
+is also a bit finicky.
+It depends on whitespace for separation of content,
+which is a natural concept for Python programmers,
+but not for most writers.
+
+In general though,
+a lot of the complexity in the language comes from the extensibility and power.
+When compared to something like Markdown,
+reStructuredText can do so much more that it's worth the complexity and somewhat steep learning curve.
+
+Introduction to Read the Docs
+-----------------------------
+
+Read the Docs is a hosting platform for Sphinx documentation.
+It takes the powerful tool that Sphinx is,
+then adds understanding of version control and other powerful features on top.
+
+Supporting Git, Mercurial, and Subversion,
+Read the Docs pulls down the code from your version control,
+then builds and host that documentation for you.
```

Comment:
"Read the Docs is a hosting platform for Sphinx-generated documentation. It takes the power of Sphinx and adds version control and (other specific examples?). It pulls down code and doc files from Git, Mercurial, and Subversion, then builds and hosts your documentation."